src/widget/wtrackmenu.cpp

```diff
@@ -630,28 +623,36 @@ TrackIdList WTrackMenu::getTrackIds() const {
         const QModelIndexList indices = getTrackIndices();
         trackIds.reserve(indices.size());
         for (const auto& index : indices) {
-            trackIds.push_back(m_pTrackModel->getTrackId(index));
+            const auto trackId = m_pTrackModel->getTrackId(index);
+            if (trackId.isValid()) {
+                trackIds.push_back(trackId);
+            }
         }
     } else {
-        const TrackPointerList trackPointers = getTrackPointers();
-        for (const auto& pTrack : trackPointers) {
+        trackIds.reserve(m_pTrackPointerList.size());
```

Comment:
Should we use the getter function for track pointers like we use the getter for track indices in `getTrackPointers()`?

Reply:
Because here it would be pointless to once again determine that we don't have an m_pTrackModel and need to access m_pTrackPointerList directly, resulting in an early return from getTrackPointers(). We have already checked that when entering the else branch.